app/controllers/application_controller.rb

```diff
@@ -23,12 +23,25 @@ def validation_error(resource)
     }, status: :bad_request
   end
 
+  def authenticate_admin
+    unless current_user.role === "admin" && current_user.is_approved
+      render json: {success: false, error: "You are not allowed to't see this page"}, status: 401
+    end
+  end
+
+  def user_is_approved
+    unless current_user.is_approved
+      render json: {success: false, error: "You can't see this page"}, status: 401
```

Comment:
I would suggest a consistency with the error message here: "You are not allowed to see this page"